src/node/element/path/data.rs

```diff
@@ -185,52 +185,71 @@ impl<'l> Parser<'l> {
 
-            'L' => Line(Absolute, parameters),
-            'l' => Line(Relative, parameters),
+            'L' => Line(Absolute, self.read_parameters(false)?.into()),
+            'l' => Line(Relative, self.read_parameters(false)?.into()),
 
-            'H' => HorizontalLine(Absolute, parameters),
-            'h' => HorizontalLine(Relative, parameters),
+            'H' => HorizontalLine(Absolute, self.read_parameters(false)?.into()),
+            'h' => HorizontalLine(Relative, self.read_parameters(false)?.into()),
 
-            'V' => VerticalLine(Absolute, parameters),
-            'v' => VerticalLine(Relative, parameters),
+            'V' => VerticalLine(Absolute, self.read_parameters(false)?.into()),
+            'v' => VerticalLine(Relative, self.read_parameters(false)?.into()),
 
-            'Q' => QuadraticCurve(Absolute, parameters),
-            'q' => QuadraticCurve(Relative, parameters),
+            'Q' => QuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            'q' => QuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'T' => SmoothQuadraticCurve(Absolute, parameters),
-            't' => SmoothQuadraticCurve(Relative, parameters),
+            'T' => SmoothQuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            't' => SmoothQuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'C' => CubicCurve(Absolute, parameters),
-            'c' => CubicCurve(Relative, parameters),
+            'C' => CubicCurve(Absolute, self.read_parameters(false)?.into()),
+            'c' => CubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'S' => SmoothCubicCurve(Absolute, parameters),
-            's' => SmoothCubicCurve(Relative, parameters),
+            'S' => SmoothCubicCurve(Absolute, self.read_parameters(false)?.into()),
+            's' => SmoothCubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'A' => EllipticalArc(Absolute, parameters),
-            'a' => EllipticalArc(Relative, parameters),
+            'A' => EllipticalArc(Absolute, self.read_parameters(true)?.into()),
+            'a' => EllipticalArc(Relative, self.read_parameters(true)?.into()),
 
             'Z' | 'z' => Close,
 
             _ => raise!(self, "found an unknown path command '{}'", name),
         }))
     }
 
-    fn read_parameters(&mut self) -> Result<Vec<Number>> {
+    fn read_parameters(&mut self, is_arc: bool) -> Result<Vec<Number>> {
         let mut parameters = Vec::new();
+        let mut cmd_index: usize = 0;
 
-        while let Some(number) = self.read_number()? {
+        while let Some(number) = self.read_number_or_flag(if is_arc { Some(cmd_index % 7) } else { None })? {
+            cmd_index = cmd_index + 1;
             parameters.push(number);
             self.reader.consume_whitespace();
             self.reader.consume_char(',');
         }
+
```

Comment:
Please remove this one.

Reply:
If I understand you correctly, you just wanted that empty line removed? If so, done.

Reply:
Yes, thank you.